platform-dom/src/main/java/com/softicar/platform/dom/elements/image/viewer/tag/DomImageViewerTagDefinitionMap.java

```diff
@@ -0,0 +1,9 @@
+package com.softicar.platform.dom.elements.image.viewer.tag;
+
+import java.util.List;
+import java.util.TreeMap;
+
+public class DomImageViewerTagDefinitionMap extends TreeMap<Integer, List<DomImageViewerTagDefinition>> {
```

Comment:
- We usually avoid extending the standard collection classes.
- If we added `addTag` methods with chaining support we might get rid of the `DomImageViewerBuilder` class?

Reply:
> We usually avoid extending the standard collection classes.
Pure convenience, I could as well delete this again.

> If we added addTag methods with chaining support we might get rid of the DomImageViewerBuilder class?
I would have preferred that as well - however, that entails the necessity of either
- calling `refresh` after each `addTag` call (that would be inefficient), or
- complicating the code to allow for ad-hoc addition of tags (and I don't think that we need that feature).

So the builder is basically there to sacrifice neither performance nor simplicity of the internal `DomImageViewer` classes.

Besides, I don't see how the two aspects you brought up are related. I could remove either of those, and retain the other.

Reply:
need to talk about this, writing is hard :)

Reply:
at least it should not extend `TreeMap` but have a field

Reply:
turned the class into `DomImageViewerTagDefinitions`. no longer extends anything.